tests/ci_build/test_r_package.py

```diff
@@ -0,0 +1,103 @@
+import argparse
+import os
+import subprocess
+
+ROOT = os.path.normpath(
+    os.path.join(os.path.dirname(os.path.abspath(__file__)), os.path.pardir,
+                 os.path.pardir))
+r_package = os.path.join(ROOT, 'R-package')
+
+
+class DirectoryExcursion:
+    def __init__(self, path: os.PathLike):
+        self.path = path
+        self.curdir = os.path.normpath(os.path.abspath(os.path.curdir))
+
+    def __enter__(self):
+        os.chdir(self.path)
+
+    def __exit__(self, *args):
+        os.chdir(self.curdir)
+
+
+def get_mingw_bin():
+    return os.path.join('c:/rtools40/mingw64/', 'bin')
+
+
+def test_with_autotools(args):
+    with DirectoryExcursion(r_package):
+        if args.compiler == 'mingw':
+            mingw_bin = get_mingw_bin()
+            CXX = os.path.join(mingw_bin, 'g++.exe')
+            CC = os.path.join(mingw_bin, 'gcc.exe')
+            cmd = ['R.exe', 'CMD', 'INSTALL', str(os.path.curdir)]
+            env = os.environ.copy()
+            env.update({'CC': CC, 'CXX': CXX})
+            subprocess.check_call(cmd, env=env)
+        elif args.compiler == 'msvc':
+            cmd = ['R.exe', 'CMD', 'INSTALL', str(os.path.curdir)]
```

Comment:
I didn't know that you could use `R CMD INSTALL` with Visual Studio compiler. Good to know.